src/Microsoft.VisualStudio.Editors/PropPages/PackagePropPage.vb

```diff
@@ -230,7 +285,117 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
+        Private Sub PackageLicenseExpression_GotFocus(sender As Object, e As EventArgs) Handles PackageLicenseExpression.GotFocus
+            SetLicenseRadioButtons(True)
+        End Sub
+
+        Private Sub LicenseFileNameTextBox_GotFocus(sender As Object, e As EventArgs) Handles LicenseFileNameTextBox.GotFocus
+            SetLicenseRadioButtons(False)
+        End Sub
+
+        Private Sub LicenseBrowseButton_GotFocus(sender As Object, e As EventArgs) Handles LicenseBrowseButton.GotFocus
+            SetLicenseRadioButtons(False)
+        End Sub
+
+        Private Shared Function GetRelativePath(fullPath As String, basePath As String) As String
+            Dim baseUri = New Uri(basePath)
+            Dim fullUri = New Uri(fullPath)
+            Dim relativeUri = baseUri.MakeRelativeUri(fullUri)
+            Return relativeUri.ToString().Replace("/", "\")
+        End Function
+
+        Private Sub AddLicenseItemToProject(fileName As String)
+            Dim unconfiguredProject = GetUnconfiguredProject(ProjectHierarchy)
+            Dim configuredProject As ProjectSystem.ConfiguredProject = ThreadHelper.JoinableTaskFactory.Run(Function()
+                                                                                                                Return unconfiguredProject.GetSuggestedConfiguredProjectAsync
+                                                                                                            End Function)
+            Dim projectSourceItemProvider = configuredProject.Services.ExportProvider.GetExportedValue(Of ProjectSystem.IProjectSourceItemProvider)()
+            Dim correctDirectory = Directory.GetParent(unconfiguredProject.FullPath).ToString
+            Dim relativePath As String = GetRelativePath(Path.GetFullPath(fileName), correctDirectory + "\")
+            LicenseFileNameTextBox.Text = relativePath
+            'The TextBox needs to have the relative path, so the property isn't linked to the TextBox. It must be set manually.
+            SetCommonPropertyValue(GetPropertyDescriptor("PackageLicenseFile"), Path.GetFileName(fileName))
+            ThreadHelper.JoinableTaskFactory.Run(Function()
+                                                     Return projectSourceItemProvider.AddAsync("None", relativePath, {(New KeyValuePair(Of String, String)("Pack", "True")), New KeyValuePair(Of String, String)("PackagePath", "")})
+                                                 End Function)
+        End Sub
+
+        Private Sub LicenseBrowseButton_Click(sender As Object, e As EventArgs) Handles LicenseBrowseButton.Click
+            Dim fileName = ""
+            Dim fileNames As ArrayList = GetFilesViaBrowse(ServiceProvider, Handle, GetProjectPath(), My.Resources.Microsoft_VisualStudio_Editors_Designer.PPG_AddExistingFilesTitle,
+                    CombineDialogFilters(
+                        "License File (*.txt, *.md)|*.txt;*.md;",
```

Comment:
Trailing semicolon is not needed here, though probably harmless.

Reply:
Does `License File` need to be localised?

Reply:
I added it as a property, so it will be localized